lib/chai/core/assertions.js

```diff
@@ -1017,16 +1017,15 @@ module.exports = function (chai, _) {
     var obj = flag(this, 'object');
     if (flag(this, 'deep')) {
       return this.eql(val);
-    } else {
```

Comment:
In this case it makes more sense to leave the `else` and remove the `return` inside the `if`
Or leave the `return` alone in a new line after calling `this.eql` and keep doing this early returning (I'm personally a big fan of early returns btw)

Thats because this function should return `undefined` and its currently sometimes returning the call of `this.eql` and sometimes just `undefined`.

I know `this.eql` will eventually return `undefined` but I need to read `this.eql` implementation to know that.

If I would refactor this code my code would look like this
Using early return:
```js
function assertEqual (val, msg) {
  if (msg) flag(this, 'message', msg);
  var obj = flag(this, 'object');
  if (flag(this, 'deep')) {
    this.eql(val);
    return;
  }

  this.assert(
      val === obj
    , 'expected #{this} to equal #{exp}'
    , 'expected #{this} to not equal #{exp}'
    , val
    , this._obj
    , true
  );
}
```
Or just removing the `return`:
```js
function assertEqual (val, msg) {
  if (msg) flag(this, 'message', msg);
  var obj = flag(this, 'object');
  if (flag(this, 'deep')) {
    this.eql(val);
  } else {
    this.assert(
        val === obj
      , 'expected #{this} to equal #{exp}'
      , 'expected #{this} to not equal #{exp}'
      , val
      , this._obj
      , true
    );
  }
}
```

Reply:
Thanks for the review.
I am also a fan of early return, so I fixed it like that.